config/v1/types_infrastructure.go

```diff
@@ -123,6 +123,60 @@ const (
 	ExternalTopologyMode TopologyMode = "External"
 )
 
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+type PlatformLoadBalancerType string
+
+const (
+	// LoadBalancerTypeUserManaged is a load balancer with control-plane VIPs managed outside of the cluster by the customer.
+	LoadBalancerTypeUserManaged PlatformLoadBalancerType = "UserManaged"
+
+	// LoadBalancerTypeOpenShiftManaged is a load balancer with control-plane VIPs managed by the OpenShift cluster.
+	LoadBalancerTypeOpenShiftManaged PlatformLoadBalancerType = "OpenShiftManaged"
+)
+
+// PlatformLoadBalancerOpenShiftManaged defines the type of the OpenShift managed load balancer used by the cluster.
+type PlatformLoadBalancerOpenShiftManagedType string
+
+const (
+	// OpenShiftManagedTypeDefault is a load balancer managed by OpenShift with default configuration (currently Keepalived + HAProxy).
+	OpenShiftManagedTypeDefault PlatformLoadBalancerOpenShiftManagedType = "Default"
+)
+
+// PlatformLoadBalancerOpenShiftManaged defines the type of the OpenShift managed load balancer used by the cluster.
+type PlatformLoadBalancerOpenShiftManaged struct {
+	// Type is the type of the OpenShift managed load balancer used by the cluster.
+	// The default value is Default.
+	// +default="Default"
+	// +kubebuilder:default:="Default"
+	// +kubebuilder:validation:Enum="Default"
+	// +kubebuilder:validation:Required
+	// +optional
+	Type PlatformLoadBalancerOpenShiftManagedType `json:"type,omitempty"`
+}
+
+// BareMetalPlatformLoadBalancer defines the load balancer used by the cluster on BareMetal platform.
+type BareMetalPlatformLoadBalancer struct {
+	// OpenShiftManaged defines the type of the OpenShift managed load balancer used by the cluster.
+	// +optional
+	OpenShiftManaged PlatformLoadBalancerOpenShiftManaged `json:"openShiftManaged,omitempty"`
+
+	// Type defines the type of load balancer used by the cluster on BareMetal platform
+	// which can be a user-managed or openshift-managed load balancer
+	// that is to be used for the OpenShift API and Ingress endpoints.
+	// When set to OpenShiftManaged the static pods in charge of API and Ingress traffic load-balancing
+	// defined in the machine config operator will be deployed.
+	// When set to UserManaged these static pods will not be deployed and it is expected that
+	// the load balancer is configured out of band by the deployer.
+	// When omitted, this means no opinion and the platform is left to choose a reasonable default.
+	// The default value is OpenShiftManaged.
+	// +default="OpenShiftManaged"
+	// +kubebuilder:default:="OpenShiftManaged"
+	// +kubebuilder:validation:Enum:="OpenShiftManaged";"UserManaged"
+	// +kubebuilder:validation:Required
+	// +optional
```

Comment:
If you want to default this, it needs to be optional, currently it's optionally required which is mildly contradictory 😉 

Please move the type field to the first field in the struct (the rest can be alphabetical) as the discriminator should be the first value in the struct